#!/usr/bin/env python
PACKAGE = "travesim_adapters"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Unicast addresses must be in the block 127.0.0.0/8. See:
# https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml
gen.add("replacer_address", str_t, 0, "Replacer Address (127.0.0.0 - 127.255.255.255)",  "127.0.0.1")

# User Ports (1024 - 49151). See:
# https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml
gen.add("replacer_port", int_t, 0, "Replacer Port", 20011,  1024, 49151)

gen.add("reset", bool_t, 0, "Reset replacer adapter",  False)
gen.add("specific_source", bool_t, 0, "Specific Source for Replacer",  False)

exit(gen.generate(PACKAGE, "replacer_adapter", "Replacer"))
